### PR TITLE
[CONTP-371]send failover metrics from node agent

### PR DESCRIPTION
**Pull request by zhuminyi**
[CONTP-371](https://datadoghq.atlassian.net/browse/CONTP-371)
### What does this PR do?

This adds the ability to send filtered series payloads to the cluster agent in local cluster. This ensures that in the rare case of an outage, the system can still collect failover metrics locally. 

also: moved GetClusterAgentEndpoint from pkg/util/clusteragent to config/utils because of go module dependency. 

### Motivation

There are limitations in integrating the cluster agent directly as a MRF endpoint, we've identified several key issues:
1. The Cluster Agent utilizes an authentication token instead of an API key.
2. The absence of an API key disrupts the domain resolver's functionality.
3. The inability to tailor the MRF filter for individual secondary endpoints can break MRF 

Proposed change:
1. A set of dedicated config options and metrics filter (supported by new _MarshalSplitCompressMultiple_)
2. Local domain resolver, Local domain forwarder and local http client for local endpoint that takes authToken 

### Describe how to test/QA your changes

Since series handler is not implemented in cluster agent yet, we can test change in local env with datadog.yaml config
```
api_key: XXXXX
telemetry.enabled: true
hostname: agent-1
autoscaling:
    failover:
        enabled: true
        metrics: datadog.agent.retry_queue_duration.bytes_per_sec
cluster_agent.enabled: true
cluster_agent.auth_token: 1234567890abcdef1234567890abcdef
cluster_agent.url: https://localhost:5002 
```
Note: we are sending series payload to https://localhost:5002/api/v2/series

Server:
Run `openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout localhost.key -out localhost.crt -subj "/C=US/ST=YourState/L=YourCity/O=YourOrganization/CN=localhost"`

Create https server with source code and run:
```
package main

import (
	"fmt"
	"net/http"
	"time"
)

func main() {
	http.HandleFunc("/api/v2/series", func(w http.ResponseWriter, r *http.Request) {
		fmt.Printf("[%s] Received request: %s %s\n", time.Now(), r.Method, r.URL.Path)
		w.WriteHeader(http.StatusOK)
		w.Write([]byte("Request received"))
	})
	http.ListenAndServeTLS(":5002", "localhost.crt", "localhost.key", nil)
}
```
Result:
```
2024-08-30 05:20:03.75237676 +0000 UTC m=+107.013497883] Received request: POST /api/v2/series
[2024-08-30 05:20:18.752380295 +0000 UTC m=+122.013776307] Received request: POST /api/v2/series
[2024-08-30 05:20:33.752624094 +0000 UTC m=+137.014020105] Received request: POST /api/v2/series
[2024-08-30 05:20:48.750682795 +0000 UTC m=+152.012268321] Received request: POST /api/v2/series
[2024-08-30 05:21:03.753627385 +0000 UTC m=+167.015212953] Received request: POST /api/v2/series
[2024-08-30 05:21:18.753528253 +0000 UTC m=+182.015410835] Received request: POST /api/v2/series
[2024-08-30 05:21:33.75313001 +0000 UTC m=+197.015012467] Received request: POST /api/v2/series
```

[CONTP-371]: https://datadoghq.atlassian.net/browse/CONTP-371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: comp/forwarder/defaultforwarder/worker.go [hunk 1]
```diff
@@ -82,6 +94,27 @@
...
+func newBearerAuthHTTPClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   1 * time.Second,
+				KeepAlive: 20 * time.Second,
+			}).DialContext,
+			ForceAttemptHTTP2:     false,
+			TLSClientConfig:       &tls.Config{InsecureSkipVerify: true},
```
[the remark this answers is not in the document]
Comment (zhuminyi) [reply comp/forwarder/defaultforwarder/worker.go:106]: for internal communication

File: pkg/config/setup/config.go [hunk 2]
```diff
@@ -671,6 +671,8 @@ func InitConfig(config pkgconfigmodel.Setup) {
 
 	// Autoscaling product
 	config.BindEnvAndSetDefault("autoscaling.workload.enabled", false)
+	config.BindEnvAndSetDefault("autoscaling.failover.enabled", false, "DD_AUTOSCALING_FAILOVER_ENABLED")
+	config.BindEnv("autoscaling.failover.metrics", "DD_AUTOSCALING_FAILOVER_METRICS")
```
Comment (pgimalac) [inline pkg/config/setup/config.go:675]: **❓ question:** ‏Is there a particular reason to specify explicitly the environment variables ? They should be used out of the box

**Review by lukesteensen — APPROVED** [review]
Looks like a reasonable extension of the MRF pattern to me 👍

**Review by dineshg13 — APPROVED** [review]
LGTM from OTel

**Review by KevinFairise2 — APPROVED** [review]

**Review by CelianR — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]

**Review by pgimalac — APPROVED** [review]
LGTM for ASC files

**Pull request merged**